src/Simplex/Chat/Library/Commands.hs

```diff
@@ -1737,38 +1737,39 @@ processChatCommand' vr = \case
+            (gInfo, hostMember) <- withStore $ \db -> createPreparedGroup db vr user groupProfile True ccLink welcomeSharedMsgId
             let cd = CDGroupRcv gInfo Nothing hostMember
-                createItem content = createInternalItemForChat user cd True content Nothing
+                createItem sharedMsgId content = createInternalItemForChat user cd True content sharedMsgId Nothing
                 cInfo = GroupChat gInfo Nothing
             void $ createGroupFeatureItems_ user cd True CIRcvGroupFeature gInfo
-            aci <- mapM (createItem . CIRcvMsgContent . MCText) message
+            aci <- mapM (createItem welcomeSharedMsgId . CIRcvMsgContent . MCText) message
             let chat = case aci of
                   Just (AChatItem SCTGroup dir _ ci) -> Chat cInfo [CChatItem dir ci] emptyChatStats {unreadCount = 1, minUnreadItemId = chatItemId' ci}
                   _ -> Chat cInfo [] emptyChatStats
             pure $ CRNewPreparedChat user $ AChat SCTGroup chat
       ACCL _ (CCLink cReq _) -> do
-        ct <- withStore $ \db -> createPreparedContact db user profile accLink
-        let createItem content = createInternalItemForChat user (CDDirectRcv ct) False content Nothing
+        ct <- withStore $ \db -> createPreparedContact db user profile accLink welcomeSharedMsgId
+        let createItem sharedMsgId content = createInternalItemForChat user (CDDirectRcv ct) False content sharedMsgId Nothing
             cInfo = DirectChat ct
-        void $ createItem $ CIRcvDirectE2EEInfo $ E2EInfo $ connRequestPQEncryption cReq
+        void $ createItem Nothing $ CIRcvDirectE2EEInfo $ E2EInfo $ connRequestPQEncryption cReq
         void $ createFeatureEnabledItems_ user ct
-        aci <- mapM (createItem . CIRcvMsgContent . MCText) message
+        aci <- mapM (createItem welcomeSharedMsgId . CIRcvMsgContent . MCText) message
         let chat = case aci of
               Just (AChatItem SCTDirect dir _ ci) -> Chat cInfo [CChatItem dir ci] emptyChatStats {unreadCount = 1, minUnreadItemId = chatItemId' ci}
               _ -> Chat cInfo [] emptyChatStats
         pure $ CRNewPreparedChat user $ AChat SCTDirect chat
   APIPrepareGroup userId ccLink groupSLinkData -> withUserId userId $ \user -> do
     let GroupShortLinkData {groupProfile = gp@GroupProfile {description}} = groupSLinkData
-    (gInfo, hostMember) <- withStore $ \db -> createPreparedGroup db vr user gp False ccLink
+    (gInfo, hostMember) <- withStore $ \db -> createPreparedGroup db vr user gp False ccLink Nothing
     let cd = CDGroupRcv gInfo Nothing hostMember
-        createItem content = createInternalItemForChat user cd True content Nothing
+        createItem content = createInternalItemForChat user cd True content Nothing Nothing
         cInfo = GroupChat gInfo Nothing
     void $ createGroupFeatureItems_ user cd True CIRcvGroupFeature gInfo
     aci <- mapM (createItem . CIRcvMsgContent . MCText) description
```

Comment:
do we check on host side that we should send group description? not sure there's a test for it